cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java

```diff
@@ -23,6 +36,61 @@ public Intellij(IdeContext context) {
     super(context, "intellij", Set.of(Tag.INTELLIJ));
   }
 
+  @Override
+  protected void runIde(String... args) {
+
+    install(true);
+
+    Step stepRun = this.context.newStep("Running IntelliJ");
+    try {
+      ProcessResult result;
+      if (this.context.getSystemInfo().isMac()) {
+        Path ideaPath = getToolPath().resolve("Contents").resolve("MacOS").resolve(IDEA);
+        result = runIntelliJ(ProcessMode.BACKGROUND,
+            CliArgument.prepend(args, "-na", ideaPath.toString(), "--args", this.context.getWorkspacePath().toString()));
+      } else {
+        result = runIntelliJ(ProcessMode.BACKGROUND, CliArgument.prepend(args, this.context.getWorkspacePath().toString()));
+      }
+      if (result.isSuccessful()) {
+        stepRun.success("Running IntelliJ successfully.");
+      } else {
+        stepRun.isFailure();
+      }
+    } catch (Exception e) {
+      stepRun.error(e, "Failed to run IntelliJ.");
+    } finally {
+      stepRun.end();
+    }
+  }
+
+  /**
+   * Runs IntelliJ application.
+   *
+   * @param processMode - the {@link ProcessMode}.
+   * @param args the individual arguments to pass to IntelliJ.
+   * @return the {@link ProcessResult}.
+   */
+  protected ProcessResult runIntelliJ(ProcessMode processMode, String... args) {
+
+    Path toolPath;
+    if (this.context.getSystemInfo().isWindows()) {
+      toolPath = getToolBinPath().resolve(IDEA64_EXE);
+    } else if (this.context.getSystemInfo().isLinux()) {
+      toolPath = getToolBinPath().resolve(IDEA_BASH_SCRIPT);
+    } else {
+      toolPath = Path.of("open");
```

Comment:
this is more a MacOS workaround and should not be a result of `getToolBinary()`. However, the `open` command needs to get the application as argument what would be the tool binary. We could even implement a global macos workaround to use `open` automatically when the ProcessContext tries to execute a MacOS `*.app`.
The current approach is rather hard to understand from the code as some parts are in `runIde` method adding magic arguments in case of MacOS and here in this method we see that `open` is used in case of MacOS. So only after reading the entire code and puzzling things together I can see that on MacOS `open -na ...somepath... --args «workspacepath» «args»` is called.